go-controller/pkg/config/config.go

```diff
@@ -658,6 +679,18 @@ func buildKubernetesConfig(exec kexec.Interface, cli, file *config, saPath strin
 		return fmt.Errorf("kubernetes API server URL scheme %q invalid", url.Scheme)
 	}
 
+	// Legacy cluster-subnet CLI option overrides config file or --k8s-cluster-subnets
+	if clusterSubnet != "" {
+		Kubernetes.RawClusterSubnets = clusterSubnet
+	}
+	if Kubernetes.RawClusterSubnets == "" {
```

Comment:
what if both clusterSubnet and Kubernetes.RawClusterSubnets are provided?
Looks like RawClusterSubnets is required...

Reply:
clusterSubnet is where the legacy CLI 'cluster-subnet' option goes. Like --service-cidr-ip-range it takes precedence over everything else.